runtime/amplitude/src/lib.rs

```diff
@@ -1287,9 +1318,10 @@ construct_runtime!(
 		Security: security::{Pallet, Call, Config, Storage, Event<T>} = 67,
 		StellarRelay: stellar_relay::{Pallet, Call, Config<T>, Storage, Event<T>} = 68,
 		VaultRegistry: vault_registry::{Pallet, Call, Config<T>, Storage, Event<T>, ValidateUnsigned} = 69,
-		VaultRewards: reward::{Pallet, Call, Storage, Event<T>} = 70,
+		VaultRewards: pooled_rewards::{Pallet, Call, Storage, Event<T>} = 70,
```

Comment:
I'm actually not 100% sure if it leads to issues if we replace this pallet with another one. Let's be on the safe side and not assign the index of `70` here but use `74`.

Reply:
But from the perspective of whatever was already using the old rewards pallet, it would look like it disappeared. Are you sure the safe side this time means changing it to 74?

Reply:
No, I'm not. I know that the order matters to some extend so I tried to look it up. In the last paragraph of [this](https://docs.substrate.io/reference/frame-macros/#runtime-construction-macros) section it is mentioned that it is important that a pallet that depends on another pallet comes after the pallet it depends on. So according to this, it would indeed be a mistake to assign `74` to the `VaultRewards` pallet since `RewardDistribution` depends on it.

What I'm still not sure about is whether we can just re-use indices. 

> But from the perspective of whatever was already using the old rewards pallet, it would look like it disappeared

I'm more concerned about the storage items of the pallet disappearing or rather _not disappearing_ and being re-used by our new pooled rewards pallet. I fear that the only way to test if the configuration you chose causes any issues is by testing the runtime upgrade locally

Reply:
Thanks for the explanation! I still don't quite understand why the storage itself would be a problem beacuse I am looking at the docs but it says right before the dependency part:

> the order used for defining pallets in the construct_runtime! macro affects the genesis storage configuration. 

In the [rust docs](https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.construct_runtime.html) it is also stated that:
> The population of the genesis storage depends on the order of pallets. 

This makes me think a runtime upgrade wouldn't be a problem I guess? Since we are not going through the genesis config and also these pallets do not have one.

Reply:
So what I am worried about is that our new pooled rewards pallet accesses storage items of the older rewards pallet that we used previously. If you look [here](https://polkadot-blockchain-academy.github.io/pba-book/frame/storage/page.html#frame-storage-keys), the pallet name is used to derive the key to the storage item in the database. And if our new pooled rewards pallet shares the same name as the old one (ie. `VaultRewards`), and it also uses the same names for some of its storage items (which it does for almost all storage items, see [here](https://github.com/pendulum-chain/spacewalk/blob/main/pallets/reward/src/lib.rs#L101-L127) vs [here](https://github.com/pendulum-chain/spacewalk/blob/main/pallets/pooled-rewards/src/lib.rs#L119-L162)), then our new pallet will query data of the old pallet which it should not. 

However it should be fine if we just rename the pallet instance in our runtime. So instead of `VaultRewards` we can call it `PooledVaultRewards` or something.

Reply:
About the indices, I think you are right and we don't need to worry about it. @b-yap found [this](https://substrate.stackexchange.com/questions/10428/reusing-old-pallet-instance-number/10429#10429) related forum post which indicates that this is not problematic as long as we don't have any pending transactions scheduled.

Reply:
So let's rename it to be on the safe side